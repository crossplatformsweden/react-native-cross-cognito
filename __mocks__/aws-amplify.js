// If the mock doesn't work, use the following in your test file:
// jest.mock('react-native-modal-datetime-picker', () => ({
//   __esModule: true,
//   default: jest.fn(),
// }));

module.exports = {
  __esModule: true,
  default: jest.fn(),
  Auth: {
    signUp: () => {
      throw new Error('No userPool');
    },
    signIn: () => {
      throw new Error('No userPool');
    },
    resendSignUp: (username) => username,
    forgotPassword: jest.fn(),
    forgotPasswordSubmit: jest.fn(),
    currentSession: jest.fn(),
    currentUserInfo: jest.fn(),
    confirmSignUp: jest.fn(),
    confirmSignIn: jest.fn(),
    signOut: jest.fn(),
    currentAuthenticatedUser: jest.fn(),
    changePassword: jest.fn(),
  },
  Analytics: {
    record: jest.fn(),
  },
};
