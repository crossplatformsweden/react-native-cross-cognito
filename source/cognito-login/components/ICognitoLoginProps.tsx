import { ICrossButtonProps } from 'react-native-cross-components';
import { ICrossEditorProps } from 'react-native-cross-components';
import { CognitoUser } from 'amazon-cognito-identity-js';
import { ReactNode } from 'react';
/**
 * Properties for the {@link CognitoLogin} component.
 *
 * Allows customization and provides the {@link onLoggedIn} event.
 *
 * Children can be supplied to the forms: {@link registerChildren}, {@link confirmChildren}, {@link forgotChildren}, {@link loginChildren}
 */
export interface ICognitoLoginProps {
  /**
   * Occurs when the user was successfully logged in and contains the resulting user.
   */
  onLoggedIn?: (user: CognitoUser) => void;

  /**
   * Optional props to customize all buttons. Typically used to change the `mode` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin buttonProps={{mode: 'contained'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  buttonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the login button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin loginButtonProps={{title: 'Engage'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  loginButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the cancel button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin cancelButtonProps={{title: 'Disengage'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  cancelButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the save button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin saveButtonProps={{title: 'Warp'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  saveButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the register button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin registerButtonProps={{title: 'Enlist'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  registerButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the phone input. Typically used to change the `label` prop or change the `maskProps`.
   * See {@link ICrossEditorProps}
   */
  phoneInputProps?: ICrossEditorProps | undefined;
  /**
   * Optional props for the e-mail input. Typically used to change the `label` prop.
   * See {@link ICrossEditorProps}
   */
  emailInputProps?: ICrossEditorProps | undefined;
  /**
   * Optional props for the password input. Typically used to change the `label` prop.
   * See {@link ICrossEditorProps}
   */
  passwordInputProps?: ICrossEditorProps | undefined;
  /**
   * Optional child components for the {@link ForgotForm} (reset password)
   */
  forgotChildren?: ReactNode;
  /**
   * Optional child components for the {@link LoginForm} (username / password)
   */
  loginChildren?: ReactNode;
  /**
   * Optional child components for the {@link RegisterForm}
   */
  registerChildren?: ReactNode;
  /**
   * Optional child components for the {@link ConfirmForm}
   */
  confirmChildren?: ReactNode;
}
