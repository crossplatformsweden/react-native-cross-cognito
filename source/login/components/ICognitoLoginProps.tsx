import { ICrossButtonProps } from 'react-native-cross-components';
import { ICrossEditorProps } from 'react-native-cross-components';
import { CognitoUser } from 'amazon-cognito-identity-js';
/**
 * Properties for the {@link CognitoLogin} component.
 *
 * Allows customization and provides the {@link onLoggedIn} event.
 */
export interface ICognitoLoginProps {
  /**
   * Occurs when the user was successfully logged in and contains the resulting user.
   */
  onLoggedIn?: (user: CognitoUser) => void;
  /**
   * Optional props for the login button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin loginButtonProps={{title: 'Engage'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  loginButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the cancel button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin cancelButtonProps={{title: 'Disengage'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  cancelButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the save button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin saveButtonProps={{title: 'Warp'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  saveButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the register button. Typically used to change the `title` prop.
   *
   * Read more:
   * https://crossplatformsweden.github.io/react-native-components/interfaces/_components_buttons_crossbutton_.icrossbuttonprops.html
   *
   * @example
   *  <CognitoLogin registerButtonProps={{title: 'Enlist'}}>
   *     <Text>Custom layouts here</Text>
   *  </CognitoLogin>
   */
  registerButtonProps?: ICrossButtonProps | undefined;
  /**
   * Optional props for the phone input. Typically used to change the `label` prop or change the `maskProps`.
   * See {@link ICrossEditorProps}
   */
  phoneInputProps?: ICrossEditorProps | undefined;
  /**
   * Optional props for the e-mail input. Typically used to change the `label` prop.
   * See {@link ICrossEditorProps}
   */
  emailInputProps?: ICrossEditorProps | undefined;
  /**
   * Optional props for the password input. Typically used to change the `label` prop.
   * See {@link ICrossEditorProps}
   */
  passwordInputProps?: ICrossEditorProps | undefined;
}