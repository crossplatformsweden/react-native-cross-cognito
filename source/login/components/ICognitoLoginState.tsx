import { ICognitoUserVariables, IAuthenticationResult } from '../../types';
import { LoginCurrentForm } from './LoginCurrentForm';
export interface ICognitoLoginState {
  /**
   * User form input. See {@link ICognitoUserVariables}
   */
  userInput: ICognitoUserVariables;
  /**
   * See {@link AuthFormState}
   */
  formState: LoginCurrentForm;
  /**
   * Result of the last authentication operation
   */
  result: IAuthenticationResult | undefined;
  code: string | undefined;
  message?: string | undefined;
}