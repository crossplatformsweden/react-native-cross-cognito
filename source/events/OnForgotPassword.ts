import { IAuthenticationResult } from '../types';
import OnCheckSession from './OnCheckSession';
import { Auth } from 'aws-amplify';

/**
 * Start the change password flow with AWS Cognito, followed by the {@link OnConfirmPassword}
 * @param input See {@link ICognitoUserVariables}
 * @returns See {@link IAuthenticationResult}
 */
export const OnForgotPassword = async (
  username: string
): Promise<IAuthenticationResult> => {
  try {
    const result = await Auth.forgotPassword(username);
    return OnCheckSession(result);
  } catch (error) {
    // if (__DEV__) console.log(error);
    return OnCheckSession(error);
  }
};

export default OnForgotPassword;
