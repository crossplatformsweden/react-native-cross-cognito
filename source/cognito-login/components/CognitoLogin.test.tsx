// https://itnext.io/testing-react-16-3-components-with-react-test-renderer-without-enzyme-d9c65d689e88
import React from 'react';
import TestRenderer from 'react-test-renderer';
import _ from 'lodash';
import { ICognitoLoginState } from './ICognitoLoginState';
import OnForgotPassword from '../../events/OnForgotPassword';
import { ICognitoLoginProps } from './ICognitoLoginProps';
import { CognitoUser, CognitoUserPool } from 'amazon-cognito-identity-js';
import { Auth } from 'aws-amplify';

// Default export requires this type of mocking
// jest.mock('react-native-stager', () => ({
//   __esModule: true,
//   Stage: 'View',
//   StageButtons: 'View',
//   StageProgress: 'View',
//   default: 'View',
// }));

describe('components', () => {
  describe('<CognitoLogin />', () => {
    beforeAll(() => {
      jest.unmock('react-native');
      jest.unmock('./CognitoLogin');

      jest.setTimeout(10000);
    });

    const GetDefaultCognitoLogin = async (
      props?: ICognitoLoginProps | undefined
    ) => {
      const { CognitoLogin } = await import('./CognitoLogin');
      const wrapper = TestRenderer.create(<CognitoLogin {...props} />);
      return wrapper;
    };

    it('Component should render', async () => {
      const wrapper = await GetDefaultCognitoLogin();
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('`state.userInput` should be defined', async () => {
      const wrapper = await GetDefaultCognitoLogin();
      expect(wrapper.root.instance.state.userInput).toBeDefined();
    });

    it('`state.userInput.email` should be `undefined`', async () => {
      const wrapper = await GetDefaultCognitoLogin();
      expect(wrapper.root.instance.state.userInput.email).toBeUndefined();
    });

    it('`state.userInput.password` should be `undefined`', async () => {
      const wrapper = await GetDefaultCognitoLogin();
      expect(wrapper.root.instance.state.userInput.password).toBeUndefined();
    });

    it('`state.userInput.formState` should be `Login`', async () => {
      const wrapper = await GetDefaultCognitoLogin();
      expect(wrapper.root.instance.state.formState).toBe('Login');
    });

    it('`loginButtonProps` should set button title', async () => {
      const wrapper = await GetDefaultCognitoLogin({
        loginButtonProps: { title: 'MyTitle' },
      });
      const child = wrapper.root.findByProps({ iconName: 'sign-in' });
      expect(child.props.title).toBe('MyTitle');
    });

    describe('Component state updates', () => {
      beforeAll(() => {
        jest.setTimeout(10000);
      });

      it('`onEmailChanged` should update state', async () => {
        const wrapper = await GetDefaultCognitoLogin();

        const expectedEmail = 'bogus@stuff.com';
        wrapper.root.instance.onEmailChanged(expectedEmail);

        expect(wrapper.root.instance.state.userInput.email).toBe(expectedEmail);
      });

      it('`onPasswordChanged` should update state', async () => {
        const wrapper = await GetDefaultCognitoLogin();

        const expectedPassword = 'sup3rSecre7';
        wrapper.root.instance.onPasswordChanged(expectedPassword);

        expect(wrapper.root.instance.state.userInput.password).toBe(
          expectedPassword
        );
      });

      it('`onPhoneChanged` should update state', async () => {
        const wrapper = await GetDefaultCognitoLogin();

        const expectedInput = '+07304655556';
        wrapper.root.instance.onPhoneChanged(expectedInput);

        expect(wrapper.root.instance.state.userInput.phone).toBe(expectedInput);
      });

      it('`onLogin` should set `state.result` to defined value', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const child = wrapper.root.findByProps({ iconName: 'sign-in' });
        await child.props.onPress();
        const result = _.get(wrapper, ['root', 'instance', 'state', 'result']);

        expect(result).not.toBeUndefined();
      });

      it('`Register` button should set `state.formState` to `Register`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const child = wrapper.root.findByProps({ title: 'Register' });
        child.props.onPress();
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);

        expect(result).toBe('Register');
      });

      it('`Forgot password` button should set `formState` to `Forgot`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const child = wrapper.root.findByProps({
          title: 'Forgot password',
        });
        child.props.onPress();
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);

        expect(result).toBe('Forgot');
      });

      it('`OnResetPassword` should set `formState` to `Login`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        await wrapper.root.instance.OnResetPassword();
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);

        expect(result).toBe('Login');
      });

      it('`ForgotPasswordAsync` should set `formState` to `NewPassword`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        await wrapper.root.instance.ForgotPasswordAsync('test@test.com');
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);

        expect(result).toBe('NewPassword');
      });
    });

    it('Error label should contain `No userPool`', async () => {
      const wrapper = await GetDefaultCognitoLogin();
      const child = wrapper.root.findByProps({ iconName: 'sign-in' });
      await child.props.onPress();

      const errorLabel = wrapper.root.findByProps({ isCaption: true });
      expect(errorLabel.props.children).toBe('No userPool');
    });

    describe('formState Register', () => {
      it('onRegister should return user', async () => {
        let result: any = undefined;
        const setUser = (newUser: any) => (result = newUser);

        const mockUser = new CognitoUser({
          Username: 'bogus@test.com',
          Pool: new CognitoUserPool({
            UserPoolId: 'us-west-1_4g646Ssds',
            ClientId: '3tsldf7voa1hol83o6tk3j2349',
          }),
        });

        const spy = jest.spyOn(Auth, 'signUp').mockImplementation(() => ({
          code: 'UserNotConfirmedException',
          user: mockUser,
        }));

        const { CognitoLogin } = await import('./CognitoLogin');

        const wrapper = TestRenderer.create(
          <CognitoLogin onRegisteredUser={setUser} />
        );
        const state: ICognitoLoginState = {
          formState: 'Register',
          code: 'testcode',
          userInput: { email: 'bogus@test.com', password: 'testPw' },
          result: undefined,
        };
        wrapper.root.instance.setState(state);
        await wrapper.root.instance.onRegister();

        expect(result).toBeDefined();
        spy.mockRestore();
      });

      it('Cancel button should set `formState` to `Login`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        wrapper.root.instance.setState({ formState: 'Register' });
        const child = wrapper.root.findByProps({ title: 'Cancel' });
        child.props.onPress();
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);

        expect(result).toBe('Login');
      });
    });

    describe('formState ConfirmAccount', () => {
      it('`onCodeChanged` should set `state.code`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        wrapper.root.instance.setState({ formState: 'ConfirmAccount' });
        const child = wrapper.root.findByProps({
          testID: 'ConfirmAccountForm',
        });
        child.props.onCodeChanged('testcode');
        const result = _.get(wrapper, ['root', 'instance', 'state', 'code']);

        expect(result).toBe('testcode');
      });

      it('`OnResendSignup` should pass', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'ConfirmAccount',
          result: undefined,
          code: undefined,
          userInput: { email: 'bogus@test.com', password: 'testPw' },
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({ title: 'Resend code' });
        expect(child.props.onPress).not.toThrow();
      });

      it('On confirm account should set `formState` to `Login`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'ConfirmAccount',
          code: '12134243',
          userInput: { email: 'bogus@test.com', password: 'testPw' },
          result: undefined,
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({
          testID: 'ConfirmAccountForm',
        });
        await child.props.onConfirmPress();
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);
        console.log('*** test result ', result);
        expect(result).toBe('Login');
        console.log('**** onConfirmPress test DONE *****');
      });

      it('When no `code` or `email` `onConfirmPress` should set `message` to `Need code and user e-mail`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'ConfirmAccount',
          code: undefined,
          userInput: { email: undefined, password: undefined },
          result: undefined,
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({
          testID: 'ConfirmAccountForm',
        });
        await child.props.onConfirmPress();
        const result = _.get(wrapper, ['root', 'instance', 'state', 'message']);
        console.log('*** test result ', result);
        expect(result).toBe('Need code and user e-mail');
        console.log('**** onConfirmPress test DONE *****');
      });
    });

    describe('formState ConfirmMFALogin', () => {
      it('`onCodeChanged` should set `state.code`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        wrapper.root.instance.setState({ formState: 'ConfirmMFALogin' });
        const child = wrapper.root.findByProps({
          testID: 'ConfirmMFAForm',
        });
        child.props.onCodeChanged('testcode');
        const result = _.get(wrapper, ['root', 'instance', 'state', 'code']);

        expect(result).toBe('testcode');
      });

      it('`onConfirmPress` should return `Not a valid Code`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'ConfirmMFALogin',
          code: 'testcode',
          userInput: { email: 'bogus@test.com', password: 'testPw' },
          result: undefined,
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({
          testID: 'ConfirmMFAForm',
        });
        await child.props.onConfirmPress();
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'result',
          'error',
          'message',
        ]);
        expect(result).toBe('Not a valid Code');
      });

      it('When no `code` or `email` `onConfirmPress` should set `message` to `Please enter username and the code`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'ConfirmMFALogin',
          code: undefined,
          userInput: { email: undefined, password: 'testPw' },
          result: undefined,
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({
          testID: 'ConfirmMFAForm',
        });
        await child.props.onConfirmPress();
        const result = _.get(wrapper, ['root', 'instance', 'state', 'message']);
        expect(result).toBe('Please enter username and the code');
      });
    });

    describe('formState NewPassword', () => {
      it('`onCodeChanged` should set `state.code`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        wrapper.root.instance.setState({ formState: 'NewPassword' });
        const child = wrapper.root.findByProps({
          testID: 'PasswordForm',
        });
        child.props.onCodeChanged('testcode');
        const result = _.get(wrapper, ['root', 'instance', 'state', 'code']);

        expect(result).toBe('testcode');
      });

      it('On save password should set formState to `Login`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'NewPassword',
          result: undefined,
          code: '12345',
          userInput: { email: 'bogus@test.com', password: 'testPw' },
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({
          testID: 'PasswordForm',
        });
        await child.props.onSavePress();
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);

        expect(result).toBe('Login');
      });

      it('When no `code` or `email` `onSavePress` should set `message` to `Need code and user e-mail`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'NewPassword',
          code: undefined,
          userInput: { email: undefined, password: undefined },
          result: undefined,
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({
          testID: 'PasswordForm',
        });
        await child.props.onSavePress();
        const result = _.get(wrapper, ['root', 'instance', 'state', 'message']);
        expect(result).toBe('Please enter username, code and new password');
      });
    });

    describe('formState Forgot', () => {
      beforeAll(() => {
        jest.setTimeout(10000);
      });

      it('When no `code` or `email` `onSavePress` should set `message` to `Need code and user e-mail`', async () => {
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'Forgot',
          code: undefined,
          userInput: { email: undefined, password: undefined },
          result: undefined,
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({
          testID: 'ForgotForm',
        });
        await child.props.onSubmit();
        const result = _.get(wrapper, ['root', 'instance', 'state', 'message']);
        expect(result).toBe('Please enter username');
      });

      it('On forgot password should not throw', async () => {
        // Have to mock the alert dialog that occurs
        jest.mock('react-native', {
          Alert: { alert: OnForgotPassword('bogus@test.com') },
        });
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'Forgot',
          result: undefined,
          code: '12345',
          userInput: { email: 'bogus@test.com', password: 'testPw' },
        };
        wrapper.root.instance.setState(state);
        const child = wrapper.root.findByProps({ testID: 'ForgotForm' });
        expect(await child.props.onSubmit).not.toThrow();
      });

      it('ForgotPassword should set formState to `NewPassword`', async () => {
        // Have to mock the alert dialog that occurs
        jest.mock('react-native', {
          Alert: jest.fn,
        });
        const wrapper = await GetDefaultCognitoLogin();
        const state: ICognitoLoginState = {
          formState: 'Forgot',
          result: undefined,
          code: '12345',
          userInput: { email: 'bogus@test.com', password: 'testPw' },
        };
        wrapper.root.instance.setState(state);
        await wrapper.root.instance.ForgotPasswordAsync('bogus@test.com');
        const result = _.get(wrapper, [
          'root',
          'instance',
          'state',
          'formState',
        ]);

        console.log('ForgotPassword result FormState: ', result);
        expect(result).toBe('NewPassword');
      });
    });
  });
});

// TODO: Test Register > Login flow (right form?)
