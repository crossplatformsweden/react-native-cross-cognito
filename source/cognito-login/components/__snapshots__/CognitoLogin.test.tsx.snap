// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <CognitoLogin /> Component should render 1`] = `
<View
  style={Object {}}
>
  <View
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        clearButtonMode="always"
        label="E-mail"
        onChangeText={[Function]}
        placeholder="E-mail"
        value="cogtest@mailinator.com"
      />
      <TextInput
        clearButtonMode="always"
        label="Password"
        onChangeText={[Function]}
        secureTextEntry={true}
        value="Password1#"
      />
    </View>
    <Button
      backgroundColor="#349b34"
      buttonStyle={
        Object {
          "margin": 0,
          "minHeight": 45,
        }
      }
      iconName="sign-in"
      mode="contained"
      onPress={[Function]}
      style={
        Object {
          "marginTop": 10,
        }
      }
      title="Log in"
    />
    <Button
      backgroundColor="#3180e0"
      buttonStyle={
        Object {
          "margin": 0,
          "minHeight": 45,
        }
      }
      mode="contained"
      onPress={[Function]}
      style={
        Object {
          "marginTop": 10,
        }
      }
      title="Register"
    />
  </View>
  <Text
    isCaption={true}
    style={
      Object {
        "color": "red",
      }
    }
  >
    
  </Text>
</View>
`;
